### DISCUSSION TITLE
TypeError: Cannot read property 'type' of undefined

### DISCUSSION THREAD

**Discussion by isaaacme** [react-hook-form/react-hook-form#2660] [Q&A] [answered]
Hi ya'll!
I'm trying to use a specific string as validation in my form, and I get  this error:
TypeError: Cannot read property 'type' of undefined

i'm attaching the error and a sandox link, thanks in advance!
  48 | passWord &&
  49 | <div>
  50 |     <label htmlFor="magicword_field">Your Magic Word</label>
> 51 |     <input type="text" id="magicword_field"
     | ^  52 |     ref={
  53 |         register({
  54 |             required: true,

https://codesandbox.io/s/exciting-bogdan-0z1bu?file=/src/App.js

**Comment by bluebill1049** [comment, marked as answer]
what's the steps to reproduce the issue?

**Reply by isaaacme** [reply]
this is the way I solved it - hope this is heplful:
```
                            ref={
                                register({
                                    required: true,
                                    validate:  value => value === 'nerd'},)} name='magicword'/>
                            {errors.magicword && errors.magicword.type === 'required' && <p className='formError'>
                                {'🔮 time to use it, just like they told you 🔮'}</p>}
                            {errors.magicword && errors.magicword.type === 'validate' && <p className='formError'>
                                {'time to stop lying to yourself, buddy!'}</p>}

```
